types/index.d.ts

```diff
@@ -0,0 +1,26 @@
+import { Client } from '@sentry/types';
+
+// add type to Vue context
+declare module 'vue/types/vue' {
+  interface Vue {
+    readonly $sentry?: Client;
+  }
+}
+
+// App Context and NuxtAppOptions
+declare module '@nuxt/types' {
+  interface Context {
+    readonly $sentry?: Client;
```

Comment:
Having `$sentry` optional, while more correct, will make using it quite inconvenient since you'll have to null check it always.

It sucks a bit...

Reply:
you are right, but I can tell that we ran into this already several times as we have turned off sentry on development..
Thanks to optional chaining this should be easily doable ;) just do this.$sentry?.captureException() (not tested, but guess this should be fine)

Reply:
Hm, sounds like a good catch, if you're saying `@nuxtjs/sentry` in types of `tsconfig.json` it means you're gonna use it and it should be defined, so it's indeed inconvenient having it optional here

Reply:
But instead of disabling you could set `initialize: false` instead.

Although if this module allows disabling (maybe it shouldn't), it wouldn't be right to advocate on making it non-optional. Unless we remove option to "disable".

Reply:
And BTW. I can't use optional chaining so I haven't switched to TS yet. For now using JS + type checking.

Reply:
well, if it would be there all the time, I would fully agree, but it is runtime dependent if it is there or not (regarding to disabled). Another solution would be to inject a $sentry "dummy" on disabled state, that just outputs a warning on all function calls that sentry is disabled and therfore not logging anything. Even though I really would like this approach, I would say this is another story / PR.

Reply:
I'm not definitely sure but I think **not having** the type optional is better

Reply:
@kevinmarrec even though I don't like it myself, I think we shouldn't make it non-optional unless the module's code is changed to make it always exist. Otherwise, we are potentially lying with types and that can trip people and cause bugs.

Reply:
Sure, so we're good with current status right ?

Reply:
I'm not happy but as far as this PR it LGTM.

Reply:
Merging now, can be improved later on :)

Reply:
please check out #136 for a possible solution :)